### PR TITLE
Add skeleton to packages page (PROJQUAY-4243)

### PR DESCRIPTION
**Pull request by HammerMeetNail**
* Fixes issue reloading pages locally that contained a tag with a `.` in the name
* Adds skeletons for Packages page [PROJQUAY-4243](https://issues.redhat.com/browse/PROJQUAY-4243)
* Handles security vulnerability timeouts on Tags page [PROJQUAY-4245](https://issues.redhat.com/browse/PROJQUAY-4245)
* Updates how unique table identifiers are calculated, makes them a little more unique
* Updates all Tag tabs to handle skeletons for three states: success, failure, pending

### CODE REVIEW

**Comment by HammerMeetNail** [comment]
The implementation for this was a little different than the one for `SecurityReport`. I don't think there will ever be an image that contains 0 packages...but if there is I can go back and account for that too.

**Comment by HammerMeetNail** [comment]
I guess maybe a `scratch` image that doesn't have any binaries? Not quite sure how we'd account for no features in the response vs a response that hasn't completed yet. I think maybe we need to track when we receive a response?

File: src/routes/RepositoryDetails/Tags/SecurityDetails.tsx [hunk 1]
```diff
@@ -92,10 +103,16 @@ export default function SecurityDetails(props: SecurityDetailsProps) {
...
+  // API call failed, hide replace skeleton with error indicator
+  if (error) {
+    return <div>Error</div>;
+  }
+
+  if (vulnCount.size) {
     return (
       <Link
         to={getTagDetailPath(props.org, props.repo, props.tag, queryParams)}
+        onClick={() => resetSecurityDetails()}
```
Comment (harishsurf) [inline src/routes/RepositoryDetails/Tags/SecurityDetails.tsx:115]: Using arrow function inside JSX hurts performance. Could this be changed to `onClick={resetSecurityDetails}`

File: src/routes/RepositoryDetails/Tags/SecurityDetails.tsx [hunk 2]
```diff
@@ -108,6 +125,7 @@ export default function SecurityDetails(props: SecurityDetailsProps) {
...
```
Comment (harishsurf) [inline src/routes/RepositoryDetails/Tags/SecurityDetails.tsx:128]: same comment as #L115. Also for #L163

File: src/routes/TagDetails/Packages/Packages.tsx [hunk 3]
```diff
@@ -1,40 +1,25 @@
...
-          const securityDetails: SecurityDetailsResponse =
-            await getSecurityDetails(props.org, props.repo, props.digest);
-          setData(securityDetails.data);
-        } catch (error) {
-          console.error('Unable to get security details: ', error);
-        }
-      })();
-    }
-  }, [props.digest]);
+  // Set features to a default of null to distinuish between a completed API call and one that is in progress
```
Comment (harishsurf) [inline src/routes/TagDetails/Packages/Packages.tsx:9]: typo: `distinguish`

File: src/routes/RepositoryDetails/Tags/Table.tsx [hunk 4]
```diff
@@ -116,7 +121,10 @@ function Row(props: RowProps) {
           }}
         />
         <Td dataLabel={columnNames.Tag}>
-          <Link to={getTagDetailPath(props.org, props.repo, tag.name)}>
+          <Link
+            to={getTagDetailPath(props.org, props.repo, tag.name)}
+            onClick={() => resetSecurityDetails()}
```
Comment (harishsurf) [inline src/routes/RepositoryDetails/Tags/Table.tsx:126]: same comment as above

File: src/routes/RepositoryDetails/Tags/SecurityDetails.tsx (hunk 1 above)
Comment (HammerMeetNail) [reply src/routes/RepositoryDetails/Tags/SecurityDetails.tsx]: TIL, fixed!

**Review by syed — APPROVED** [review]
LGTM

**Pull request merged**